sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java

```diff
@@ -250,6 +244,18 @@ public String getId() {
         }
 
         final CosmosItemRequestOptions requestOptions = options;
+
+        if (this.bulkExecutionEnabled) {
+            @SuppressWarnings("unchecked")
+            Class<T> itemType = (Class<T>) item.getClass();
+            return bulkExecutionInternal(ModelBridgeInternal.toRequestOptions(requestOptions),
+                item,
+                null,
```

Comment:
If id is provided in the to-be-created document, shouldn't it be passed down here as well?

Reply:
Then we have to parse the document to get the id right?, and it's not needed here as such.